src/Nethermind/Nethermind.Network/SnapCapabilitySwitcher.cs

```diff
@@ -27,29 +28,30 @@ namespace Nethermind.Network;
 public class SnapCapabilitySwitcher
 {
     private readonly IProtocolsManager _protocolsManager;
-    private readonly ProgressTracker _progressTracker;
+    private readonly MultiSyncModeSelector _multiSyncModeSelector;
+    private readonly ILogger _logger;
 
-    public SnapCapabilitySwitcher(IProtocolsManager? protocolsManager, ProgressTracker? progressTracker)
+    public SnapCapabilitySwitcher(IProtocolsManager? protocolsManager, MultiSyncModeSelector? multiSyncModeSelector, ILogManager? logManager)
```

Comment:
NP: I'd add `SyncProgressResolver` instead.

Reply:
Probably make things worst though.

Reply:
We need an event when state sync will be finished to be able to remove `snap` capability. There is no such event in `SyncProgressResolver` so we need to stay with `MultiSyncModeSelector`